app/src/main/java/com/sq/thed_ck_licker/Game.kt

```
@@ -33,17 +41,32 @@ fun Game(innerPadding: PaddingValues) {
     val playerCardCount = rememberSaveable { mutableIntStateOf(0) }
     val latestCard = rememberSaveable { mutableIntStateOf(-1) }
     val playerHealth =
-        rememberSaveable { TheGameHandler.getPlayerHealthM() }
-    val playerScore = rememberSaveable { TheGameHandler.getPlayerScoreM() }
-    // TODO here probably should be viewModel from about the game state data or something like that
-    //  Something about state holder and all that
+        rememberSaveable { TheGameHandler.playerSystem.getPlayerHealthM() }
+    val playerScore = rememberSaveable { TheGameHandler.playerSystem.getPlayerScoreM() }
+    // TODO we should read more about viewModels and state holding
 
     val modifier = Modifier
 
     val activateCard = {
+        // TODO: I now think all these should be such as the one bellow, just nice clean function calls
+        onTurnStartEffectStackSystem()
         cardsSystem.activateCard(latestCard, playerCardCount)
+        onDeathSystem()
     }
+
     val pullNewCard = {
+        try {
+            (latestCard.intValue get EffectComponent::class).onDeactivate.invoke(getPlayerID())
+        } catch (_: Exception) {
+            println("Yeah yeah, we get it, you are so cool there was no effect component ")
```

I think we should maybe throw an exception here for easer tesing later.

---

no

---

And to be more precise, It is because we dont want actual error as long as there are new and old way cards in system.
All the old ways will fail this always.
But the new ones need it.